core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -433,8 +433,10 @@ private[spark] class DAGScheduler(
    * submission.
    */
   private def checkBarrierStageWithNumSlots(rdd: RDD[_]): Unit = {
-    if (rdd.isBarrier() && rdd.getNumPartitions > sc.maxNumConcurrentTasks) {
-      throw new BarrierJobSlotsNumberCheckFailed
+    lazy val numPartitions = rdd.getNumPartitions
```

Comment:
Oh wait, one last little thing: why are these local vars lazy? They're cheap to access. Even if not, just nest two if statements below to conditionally access them.

Reply:
Yes they, especially `rdd.getNumPartitions`, are cheap to access. It's just for removing  unused variable in case `rdd.isBarrier()` returns `false`.
I have no strong preference for whether making them lazy, nesting if statements or just removing `lazy`  modifier.

It's just my curiosity but making those variables lazy brings another overhead or pool readability of code?

Anyway, I'll remove `lazy`

Reply:
I'm actually not sure what if anything it does for a local. It introduces another lambda/function and check when accessed though, so probably isn't even saving anything.

Reply:
Thanks, I understand.